CONTRIBUTING.md

```diff
@@ -0,0 +1,337 @@
+Contributing Guidelines
+=======================
+
+Before diving into the codebase, it's highly recommended that you have some basic knowledge
+about how OOP works in C++. Some concepts that you should know include inheritance, polymorphism,
+virtual members, etc. Knowledge of the C++ Standard Template Library (STL) is also helpful. [Effective
+STL](https://www.amazon.ca/Effective-STL-Specific-Standard-Template/dp/0201749629) is a great book
+that covers useful STL concepts in depth.
+
+---
+
+Style Guide
+-----------
+> An slightly modified excerpt from [C++ Best Practices](https://www.gitbook.com/book/sadmansk/cpp-best-practices/details)
+
+
+Consistency is the most important aspect of style. The second most important aspect is following a style that the average C++ programmer is used to reading.
+
+C++ allows for arbitrary-length identifier names, so there's no reason to be terse when naming things. Use descriptive names, and be consistent in the style.
+
+ * `CamelCase`
+ * `snake_case`
+
+are common examples. *snake_case* has the advantage that it can also work with spell checkers, if desired.
+
+### Common C++ Naming Conventions
+
+ * Types start with upper case: `MyClass`.
+ * Functions and variables start with lower case: `myMethod`.
+ * Constants are all upper case: `const double PI=3.14159265358979323;`.
```

Comment:
For styling, can you add spaces before/after the `=`? Just so it's easier to read and I believe we usually add them anyways.